### DISCUSSION TITLE
Can I reinstall one of agent nodes?

### DISCUSSION THREAD

**Discussion by maximen39** [mysticaltech/terraform-hcloud-kube-hetzner#1070] [Q&A]
Hello! I recently created a cluster and it worked fine until one of the nodes stopped working. It simply does not connect to the server, the private network is not picked up, and rescue mode does not work. Hetzner's tech support said that it can be reinstalled, or installed from a snapshot, if available. The question is, can I reinstall the agent, or delete it correctly and add another one?

**Comment by otavio** [comment]
Same happened here. The node is totally broken, so how can I force it to be recreated?

**Comment by andi0b** [comment]
> **⚠️ Please try this only for a test-cluster first, and not on a production cluster, I don't know if it actually works! ⚠️**

I would try to delete the server with the Hetzner Console and then run Terraform again, which should re-create it. The node should be drained before off course, and any local data backuped.
